source/backend/dal/Helpers/Extensions/PropertyViewExtensions.cs

```diff
@@ -54,22 +51,27 @@ public static class PropertyViewExtensions
         /// <param name="user"></param>
         /// <param name="filter"></param>
         /// <returns></returns>
-        private static ExpressionStarter<PimsPropertyVw> GenerateCommonPropertyQuery(PimsContext context, ClaimsPrincipal user, Entity.Models.PropertyFilter filter)
+        private static ExpressionStarter<PimsPropertyVw> GenerateCommonPropertyQuery(PimsContext context, ClaimsPrincipal user, Entities.Models.PropertyFilter filter)
         {
-            filter.ThrowIfNull(nameof(filter));
-            filter.ThrowIfNull(nameof(user));
-
-            // Check if user has the ability to view sensitive properties.
-            var viewSensitive = user.HasPermission(Permissions.SensitiveView);
+            ArgumentNullException.ThrowIfNull(filter, nameof(filter));
+            ArgumentNullException.ThrowIfNull(user, nameof(user));
 
             var predicateBuilder = PredicateBuilder.New<PimsPropertyVw>(p => true);
 
-            if (!string.IsNullOrWhiteSpace(filter.PinOrPid))
+            if (!string.IsNullOrWhiteSpace(filter.Pid))
+            {
+                // note - 2 part search required. all matches found by removing leading 0's, then matches filtered in subsequent step. This is because EF core does not support an lpad method.
+                Regex nonInteger = new Regex("[^\\d]");
+                var formattedPid = Convert.ToInt32(nonInteger.Replace(filter.Pid, string.Empty)).ToString();
+                predicateBuilder = predicateBuilder.And(p => EF.Functions.Like(p.Pid.ToString(), $"%{formattedPid}%"));
+            }
+
+            if (!string.IsNullOrWhiteSpace(filter.Pin))
             {
                 // note - 2 part search required. all matches found by removing leading 0's, then matches filtered in subsequent step. This is because EF core does not support an lpad method.
```

Comment:
this comment is no longer applicable here, correct?

Reply:
Not sure if it applies to be honest. I think you implemented the original logic and I didn't want to break it.
If you say that the padding applies to PID only then I will remove the 2-part search from the PIN option (leaving PID as-is)